node/src/actors/chain_manager/actor.rs

```diff
@@ -177,4 +175,22 @@ impl ChainManager {
             })
             .wait(ctx);
     }
+
+    /// Load public key hash from signature manager
+    fn get_pkh(&mut self, ctx: &mut Context<Self>) {
+        signature_mngr::pkh()
+            .into_actor(self)
+            .map_err(|e, _act, _ctx| {
+                error!(
+                    "Error while getting public key hash from signature manager: {}",
+                    e
+                );
+            })
+            .and_then(|res, act, _ctx| {
+                act.own_pkh = Some(res);
+                debug!("Public key hash successfully loaded from signature manager");
+                actix::fut::ok(())
+            })
+            .wait(ctx);
```

Comment:
*Suggestion*: the wait can be avoided if *"in principle"* we remove the race condition of changing the actor state from inside the callback